Wikipedia/Code/WMFDatabaseHouseKeeper.swift

```diff
@@ -8,9 +8,30 @@ import Foundation
         let urls = try deleteStaleUnreferencedArticles(moc, navigationStateController: navigationStateController)
 
         try deleteStaleTalkPages(moc)
+        try deleteStaleAnnouncements(moc)
 
         return urls
     }
+    
+    private func deleteStaleAnnouncements(_ moc: NSManagedObjectContext) throws {
+        let predicate = NSPredicate(format: "isVisible == NO")
```

Comment:
Feel free to ignore if I'm misunderstanding, but is this particular check for `isVisible == NO` necessary? If the goal is to clean up "stale" no longer relevant announcements, doesn't just using all the announcements in the for loop guard below, removing any announcement that has an end date before the current date, cover what we're trying to accomplish (whether or not it's visible)?

I only bring this up because if there's some unexpected `isVisible` toggling logic elsewhere, it would potentially prevent the stale announcements in the upcoming for loop from being purged.

Reply:
@staykids hm yeah that is a fair point - I think my reasoning was to be a little more light touch with this cleanup since it was more of a last-minute change. But you're right that if we can't rely on `isVisible` to be entirely accurate, maybe we should remove that consideration from this cleanup. I'll push an update with that fix and will do some testing on my end with it.